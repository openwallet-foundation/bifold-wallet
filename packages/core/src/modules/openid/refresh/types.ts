import {
  ClaimFormat,
  JwaSignatureAlgorithm,
  JwkJson,
  MdocRecord,
  SdJwtVcRecord,
  W3cCredentialRecord,
} from '@credo-ts/core'
import { OpenId4VciResolvedCredentialOffer } from '@credo-ts/openid4vc'

export type IssuerMetadataCache = {
  credential_issuer: string
  credential_endpoint?: string
  token_endpoint?: string
  authorization_servers?: string[]
  credential_configurations_supported?: Record<string, any>
  // optionally: added_at, etag, updated_at, etc.
}

export enum RefreshStatus {
  Valid = 'valid',
  Invalid = 'invalid',
  Error = 'error',
}

export interface RefreshCredentialMetadata {
  // issuer & config to re-issue the same cred
  authServer: string
  tokenEndpoint: string
  refreshToken: string
  credentialIssuer: string
  credentialConfigurationId: string
  issuerMetadataCache: IssuerMetadataCache
  clientId?: string
  tokenBinding?: 'DPoP' | 'Bearer'
  dpop?: { alg: JwaSignatureAlgorithm; jwk: JwkJson }

  // refresh lifecycle
  lastCheckedAt?: number
  lastCheckResult?: RefreshStatus
  attemptCount?: number
  nextEligibleAt?: number

  // linking (optional)
  supersededByID?: string
  supersedesID?: string
  resolvedCredentialOffer?: OpenId4VciResolvedCredentialOffer
}

export type RefreshOrchestratorOpts = {
  intervalMs?: number | null
  autoStart?: boolean
  onError?: (e: unknown) => void
  listRecords?: () => Promise<any[]>
  toLite?: (rec: W3cCredentialRecord | SdJwtVcRecord | MdocRecord) => {
    id: string
    format: ClaimFormat
    createdAt?: string
    issuer?: string
  }
}

export interface IRefreshOrchestrator {
  configure(opts: Partial<RefreshOrchestratorOpts>): void
  start(): void
  stop(): void
  runOnce(reason?: string): Promise<void>
  isRunning(): boolean
  resolveFull(id: string): W3cCredentialRecord | SdJwtVcRecord | MdocRecord | undefined
}

export enum OpenIDCustomNotificationType {
  CredentialReplacementAvailable = 'CustomNotificationOpenIDCredential',
}
