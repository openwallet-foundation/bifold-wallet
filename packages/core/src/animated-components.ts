import ButtonLoading from './components/animated/ButtonLoading'
import ConnectionLoading from './components/animated/ConnectionLoading'
import CredentialAdded from './components/animated/CredentialAdded'
import CredentialPending from './components/animated/CredentialPending'
import LoadingIndicator from './components/animated/LoadingIndicator'
import RecordLoading from './components/animated/RecordLoading'
import SendingProof from './components/animated/SendingProof'
import SentProof from './components/animated/SentProof'

export interface AnimatedComponents {
  ButtonLoading: React.FC
  ConnectionLoading: React.FC
  CredentialAdded: React.FC
  CredentialPending: React.FC
  LoadingIndicator: React.FC
  RecordLoading: React.FC
  SendingProof: React.FC
  SentProof: React.FC
}

export const animatedComponents: AnimatedComponents = {
  ButtonLoading: ButtonLoading,
  ConnectionLoading: ConnectionLoading,
  CredentialAdded: CredentialAdded,
  CredentialPending: CredentialPending,
  LoadingIndicator: LoadingIndicator,
  RecordLoading: RecordLoading,
  SendingProof: SendingProof,
  SentProof: SentProof,
}
