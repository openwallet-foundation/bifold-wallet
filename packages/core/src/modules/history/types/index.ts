import { GenericRecord } from '@credo-ts/core/build/modules/generic-records/repository/GenericRecord'
import { Moment } from 'moment'

export interface IHistoryManager {
  saveHistory: (recordData: HistoryRecord) => Promise<void>
  getHistoryItems: (query: HistoryQuery) => Promise<CustomRecord[]>
  findGenericRecordById: (id: string) => Promise<GenericRecord | null>
  removeGenericRecord: (genericRecord: GenericRecord) => Promise<void>
  handleStoreHistorySettings: (selectedValue: HistoryBlockSelection | undefined) => Promise<void>
  getStoredHistorySettingsOption: () => Promise<string | null>
  getHistorySettingsOptionList: () => Array<HistoryBlockSelection>
}

export interface HistoryBlockSelection {
  value: string
  id: string
  key?: number
  date?: Moment
}

export enum HistoryCardType {
  CardAccepted = 'CardAccepted',
  CardDeclined = 'CardDeclined',
  CardExpired = 'CardExpired', // TODO: log this type of event
  CardRemoved = 'CardRemoved',
  CardRevoked = 'CardRevoked', // TODO: log this type of event
  CardUpdates = 'CardUpdates', // TODO: log this type of event
  PinChanged = 'PinChanged',
  InformationSent = 'InformationSent',
  InformationNotSent = 'InformationNotSent',
  Connection = 'Connection',
  ConnectionRemoved = 'ConnectionRemoved',
  ActivateBiometry = 'ActivateBiometry',
  DeactivateBiometry = 'DeactivateBiometry',
}
export interface HistoryRecord {
  /** History Record ID */
  id?: string
  /** History Record Type */
  type?: HistoryCardType
  /** History Record Message */
  message: string
  /** History Record Date */
  createdAt?: Date
  /** History Record Item ID */
  correspondenceId?: string
  /** History Record Item Name */
  correspondenceName?: string
}
export enum RecordType {
  AppNotificationRecord = 'AppNotificationRecord',
  NotificationRecord = 'NotificationRecord',
  HistoryRecord = 'HistoryRecord',
}
export type TagObject = {
  [key: string]: string | boolean | undefined | Array<string> | null
  [key: number]: never
}

export interface CustomRecord {
  content: RecordTypes
  tags?: TagObject
}

export interface AppNotificationRecord {
  id?: string
  createdAt?: Date
  type?: AppNotificationCardType
}

export interface NotificationRecord {
  message: string
  credentialId: string
  type?: NotificationCardType
  id?: string
  createdAt?: Date
  isNotificationRemoved: boolean
}

export enum AppNotificationCardType {
  AppPinReminder = 'AppPinReminder',
}

export enum NotificationCardType {
  CardExpired = 'CardExpired',
  CardRevoked = 'CardRevoked',
}

type RecordTypes = NotificationRecord | HistoryRecord | AppNotificationRecord
export type HistoryQuery = {
  type: RecordType
}
