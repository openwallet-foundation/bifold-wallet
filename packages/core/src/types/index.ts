import * as state from './state'
export { state }
