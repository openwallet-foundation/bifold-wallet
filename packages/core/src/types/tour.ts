import { ReactNode } from 'react'

/**
 * Makes the target optional. I.e, it can be the target type or `undefined`.
 */
export type Optional<T> = T | undefined

/**
 * An alias of what a React child looks when passed as function.
 */
export type ChildFn<T> = (value: T) => ReactNode

export enum BaseTourID {
  HomeTour = 'homeTourSteps',
  CredentialsTour = 'credentialsTourSteps',
  CredentialOfferTour = 'credentialOfferTourSteps',
  ProofRequestTour = 'proofRequestTourSteps',
}

export type TourID = `${string}TourSteps`
