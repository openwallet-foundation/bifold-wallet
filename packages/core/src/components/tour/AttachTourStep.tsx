import React, { cloneElement, ReactElement, ReactNode, RefObject, useContext, useEffect, useRef } from 'react'
import { StyleProp, useWindowDimensions, View } from 'react-native'

import { TourContext } from '../../contexts/tour/tour-context'
import { TourID } from '../../types/tour'
import { testIdWithKey } from '../../utils/testable'

export interface ChildProps<T> {
  children?: ReactNode
  ref: RefObject<unknown>
  style: StyleProp<T>
}

export interface AttachTourStepProps<T> {
  children: ReactElement<ChildProps<T>>
  /**
   * When `AttachTourStep` wraps a Functional Component, it needs to add an
   * aditional `View` on top of it to be able to measure the layout upon
   * render. This prop allows to define the behavior of the width of such
   * `View`. When set to `false`, it adjusts to its contents, when set to
   * `true`, it stretches out and tries to fill it view.
   *
   * **Note:** This prop has no effect when wrapping native components or
   * componentes created with `React.forwardRef`, which pass the `ref` to
   * another native component.
   *
   * @default false
   */
  fill?: boolean
  /**
   * The index of the `steps` array to which the step is attatched to.
   */
  index: number
  /**
   * The id of the tour this step belongs to (ie. home tour or proof request tour or etc.)
   */
  tourID: TourID
}

/**
 * React functional component used to attach and step to another component by
 * only wrapping it. Use its props to customize the behavior.
 *
 * @param props the component props
 * @returns an AttachTourStep React element
 */
export function AttachTourStep<T>({ children, fill = false, index, tourID }: AttachTourStepProps<T>): ReactElement {
  const { currentStep, currentTour, changeSpot } = useContext(TourContext)
  const { width: windowWidth, height: windowHeight } = useWindowDimensions()
  const childRef = useRef<View>(null)

  useEffect(() => {
    if (currentTour === tourID && currentStep === index) {
      childRef.current?.measureInWindow((x, y, width, height) => {
        changeSpot({ height, width, x, y })
      })
    }
  }, [currentTour, tourID, currentStep, index, windowWidth, windowHeight, changeSpot])

  const { style, ...rest } = children.props
  const childStyle = style ?? {}
  return (
    <View
      testID={testIdWithKey('AttachTourStep')}
      ref={childRef}
      style={{ alignSelf: fill ? 'stretch' : 'center', ...childStyle }}
      collapsable={false}
      focusable={false}
    >
      {cloneElement(children, rest, children.props?.children)}
    </View>
  )
}
