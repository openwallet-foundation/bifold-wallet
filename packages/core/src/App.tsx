import React, { useEffect } from 'react'
import { StatusBar } from 'react-native'
import SplashScreen from 'react-native-splash-screen'
import Toast from 'react-native-toast-message'

import { useNavigationContainerRef } from '@react-navigation/native'
import { isTablet } from 'react-native-device-info'
import Orientation from 'react-native-orientation-locker'
import { animatedComponents } from './animated-components'
import ErrorModal from './components/modals/ErrorModal'
import toastConfig from './components/toast/ToastConfig'
import { tours } from './constants'
import { Container, ContainerProvider } from './container-api'
import { AnimatedComponentsProvider } from './contexts/animated-components'
import { AuthProvider } from './contexts/auth'
import NavContainer from './contexts/navigation'
import { NetworkProvider } from './contexts/network'
import { StoreProvider } from './contexts/store'
import { ThemeProvider } from './contexts/theme'
import { TourProvider } from './contexts/tour/tour-provider'
import { initStoredLanguage } from './localization'
import RootStack from './navigators/RootStack'
import { bifoldTheme, themes } from './theme'

const createApp = (container: Container): React.FC => {
  const AppComponent: React.FC = () => {
    const navigationRef = useNavigationContainerRef()

    useEffect(() => {
      initStoredLanguage().then()
    }, [])

    useEffect(() => {
      // Hide the native splash / loading screen so that our
      // RN version can be displayed.
      SplashScreen.hide()
    }, [])

    if (!isTablet()) {
      Orientation.lockToPortrait()
    }

    return (
      <ContainerProvider value={container}>
        <StoreProvider>
          <ThemeProvider themes={themes} defaultThemeName={bifoldTheme.themeName}>
            <NavContainer navigationRef={navigationRef}>
              <AnimatedComponentsProvider value={animatedComponents}>
                <AuthProvider>
                  <NetworkProvider>
                    <StatusBar
                      hidden={false}
                      barStyle="light-content"
                      backgroundColor={bifoldTheme.ColorPalette.brand.primary}
                      translucent={false}
                    />
                    <ErrorModal />
                    <TourProvider tours={tours} overlayColor={'gray'} overlayOpacity={0.7}>
                      <RootStack />
                    </TourProvider>
                    <Toast topOffset={15} config={toastConfig} />
                  </NetworkProvider>
                </AuthProvider>
              </AnimatedComponentsProvider>
            </NavContainer>
          </ThemeProvider>
        </StoreProvider>
      </ContainerProvider>
    )
  }

  return AppComponent
}

export default createApp
