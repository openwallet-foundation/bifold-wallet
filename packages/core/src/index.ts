/* eslint-disable import/no-cycle */
import type { OnboardingStyleSheet } from './screens/Onboarding'

import { Agent } from '@credo-ts/core'
import AgentProvider from '@credo-ts/react-hooks'

import createApp from './App'
import * as components from './components'
import { ButtonImpl as Button, ButtonType, Button as IButton } from './components/buttons/Button'
import IconButton, { ButtonLocation } from './components/buttons/IconButton'
import BulletPoint from './components/inputs/BulletPoint'
import CheckBoxRow from './components/inputs/CheckBoxRow'
import LimitedTextInput from './components/inputs/LimitedTextInput'
import NotificationListItem from './components/listItems/NotificationListItem'
import ContentGradient from './components/misc/ContentGradient'
import CredentialCard from './components/misc/CredentialCard'
import ErrorBoundaryWrapper from './components/misc/ErrorBoundary'
import FauxHeader from './components/misc/FauxHeader'
import InfoBox, { InfoBoxType } from './components/misc/InfoBox'
import QRScannerTorch from './components/misc/QRScannerTorch'
import DeveloperModal from './components/modals/DeveloperModal'
import ErrorModal from './components/modals/ErrorModal'
import SafeAreaModal from './components/modals/SafeAreaModal'
import Record from './components/record/Record'
import InfoTextBox from './components/texts/InfoTextBox'
import Link from './components/texts/Link'
import Text from './components/texts/Text'
import { ThemedText } from './components/texts/ThemedText'
import { ToastType } from './components/toast/BaseToast'
import toastConfig from './components/toast/ToastConfig'
import { AttachTourStep } from './components/tour/AttachTourStep'
import { credentialOfferTourSteps } from './components/tour/CredentialOfferTourSteps'
import { credentialsTourSteps } from './components/tour/CredentialsTourSteps'
import { homeTourSteps } from './components/tour/HomeTourSteps'
import { proofRequestTourSteps } from './components/tour/ProofRequestTourSteps'
import { TourBox } from './components/tour/TourBox'
import { Banner, BannerMessage, BannerSection } from './components/views/Banner'
import HomeFooterView from './components/views/HomeFooterView'
import KeyboardView from './components/views/KeyboardView'
import { attemptLockoutConfig, PINRules, tours, walletTimeout } from './constants'
import { defaultConfig, defaultHistoryEventsLogger } from './container-impl'
import * as contexts from './contexts'
import { ActivityProvider, AutoLockTime, useActivity } from './contexts/activity'
import { AuthProvider } from './contexts/auth'
import NavContainer from './contexts/navigation'
import { NetworkProvider } from './contexts/network'
import { useTour } from './contexts/tour/tour-context'
import { TourProvider } from './contexts/tour/tour-provider'
import { useDeveloperMode } from './hooks/developer-mode'
import usePreventScreenCapture from './hooks/screen-capture'
import useBifoldAgentSetup from './hooks/useBifoldAgentSetup'
import { OpenIDCredentialRecordProvider } from './modules/openid/context/OpenIDCredentialRecordProvider'
import { DefaultScreenLayoutOptions } from './navigators/defaultLayoutOptions'
import { DefaultScreenOptionsDictionary, useDefaultStackOptions } from './navigators/defaultStackOptions'
import AttemptLockout from './screens/AttemptLockout'
import Biometry from './screens/Biometry'
import Developer from './screens/Developer'
import Onboarding from './screens/Onboarding'
import OnboardingPages from './screens/OnboardingPages'
import Preface from './screens/Preface'
import Scan from './screens/Scan'
import Splash from './screens/Splash'
import Terms from './screens/Terms'
import UpdateAvailable from './screens/UpdateAvailable'
import { bifoldLoggerInstance } from './services/bifoldLogger'
import { isBiometricsActive, loadLoginAttempt } from './services/keychain'
import { BifoldLogger } from './services/logger'
import { AbstractBifoldLogger } from './services/AbstractBifoldLogger'
import { DeepPartial, ThemeBuilder } from './theme-builder'
import * as types from './types'
import { CredentialListFooterProps } from './types/credential-list-footer'

export { animatedComponents } from './animated-components'
export { EventTypes, LocalStorageKeys } from './constants'
export { AnimatedComponentsProvider, useAnimatedComponents } from './contexts/animated-components'
export { useAuth } from './contexts/auth'
export { useNetwork } from './contexts/network'
export { DispatchAction, reducer, default as Store } from './contexts/reducers/store'
export { defaultState, mergeReducers, StoreContext, StoreProvider, useStore } from './contexts/store'
export { ThemeProvider, useTheme } from './contexts/theme'
export { useDeepLinks } from './hooks/deep-links'
export { initLanguages, initStoredLanguage, Locales, translationResources } from './localization'
export * from './navigators'
export { createStyles } from './screens/OnboardingPages'
export * from './services/storage'
export { bifoldTheme, ColorPalette, Assets as ImageAssets } from './theme'
export * from './types/attestation'
export { BifoldError } from './types/error'
export { Screens, Stacks, TabStacks } from './types/navigators'
export * from './types/version-check'
export { createLinkSecretIfRequired, getAgentModules } from './utils/agent'
export { getCredentialIdentifiers, isValidAnonCredsCredential } from './utils/credential'
export {
  connectFromScanOrDeepLink,
  formatTime,
  getConnectionName,
  removeExistingInvitationsById,
  useCredentialConnectionLabel,
} from './utils/helpers'
export { getIndyLedgers, IndyLedger, readIndyLedgersFromFile, writeIndyLedgersToFile } from './utils/ledger'
export { statusBarStyleForColor, StatusBarStyles } from './utils/luminance'
export { migrateToAskar } from './utils/migration'
export { buildFieldsFromAnonCredsCredential } from './utils/oca'
export { testIdForAccessabilityLabel, testIdWithKey } from './utils/testable'

export type { AnimatedComponents } from './animated-components'
export type { ReducerAction } from './contexts/reducers/store'
export type { RenderProps, TourStep } from './contexts/tour/tour-context'
export type {
  IAssets,
  IBrandColors,
  IColorPalette,
  IErrorColors,
  IFontAttributes,
  IGrayscaleColors,
  IInlineInputMessage,
  IInputAttributes,
  IInputs,
  INotificationColors,
  ISemanticColors,
  ISpacing,
  ISVGAssets,
  ITabTheme,
  ITextTheme,
  ITheme,
} from './theme'
export type { GenericFn } from './types/fn'
export { BasicMessageMetadata, CredentialMetadata } from './types/metadata'
export type { basicMessageCustomMetadata, credentialCustomMetadata } from './types/metadata'
export type { ContactStackParams, NotificationStackParams, OnboardingStackParams } from './types/navigators'
export type { WalletSecret } from './types/security'
export type {
  LoginAttempt as LoginAttemptState,
  Migration as MigrationState,
  Onboarding as OnboardingState,
  PersistentState,
  Preferences as PreferencesState,
  State,
  Tours as ToursState,
  VersionInfo,
} from './types/state'
export type { BifoldAgent } from './utils/agent'
export type { IndyLedgerConfig, IndyLedgerFileSystem, IndyLedgerJSON, IndyLedgersRecord } from './utils/ledger'
export type { OnboardingStyleSheet }

export type { InlineMessageProps } from './components/inputs/InlineErrorText'

export { InlineErrorPosition } from './types/error'

export * from './container-api'
export { MainContainer } from './container-impl'
export type { CredentialListFooterProps }

export type { Config, HistoryEventsLoggerConfig } from './types/config'
export type { OnboardingTask, ScreenLayoutConfig } from './types/navigators'

export { LockoutReason } from './contexts/auth'
export type { OnboardingStackProps } from './navigators/OnboardingStack'
export type { SplashProps } from './screens/Splash'
export { BaseTourID } from './types/tour'

export type { BannerSectionProps } from './components/views/Banner'

export {
  ActivityProvider,
  Agent,
  AgentProvider,
  AttachTourStep,
  AttemptLockout,
  attemptLockoutConfig,
  AuthProvider,
  AutoLockTime,
  Banner,
  BannerSection,
  AbstractBifoldLogger,
  BifoldLogger,
  bifoldLoggerInstance,
  Biometry,
  BulletPoint,
  Button,
  ButtonLocation,
  ButtonType,
  CheckBoxRow,
  components,
  ContentGradient,
  contexts,
  createApp,
  CredentialCard,
  credentialOfferTourSteps,
  credentialsTourSteps,
  defaultConfig,
  defaultHistoryEventsLogger,
  DefaultScreenLayoutOptions,
  DefaultScreenOptionsDictionary,
  Developer,
  DeveloperModal,
  ErrorBoundaryWrapper,
  ErrorModal,
  FauxHeader,
  HomeFooterView as HomeContentView,
  homeTourSteps,
  IconButton,
  InfoBox,
  InfoBoxType,
  InfoTextBox,
  isBiometricsActive,
  KeyboardView,
  LimitedTextInput,
  Link,
  loadLoginAttempt,
  NavContainer,
  NetworkProvider,
  NotificationListItem,
  Onboarding,
  OnboardingPages,
  OpenIDCredentialRecordProvider,
  PINRules,
  Preface,
  proofRequestTourSteps,
  QRScannerTorch,
  Record,
  SafeAreaModal,
  Scan,
  Splash,
  Terms,
  Text,
  ThemeBuilder,
  ThemedText,
  toastConfig,
  ToastType,
  TourBox,
  TourProvider,
  tours,
  types,
  UpdateAvailable,
  useActivity,
  useBifoldAgentSetup,
  useDefaultStackOptions,
  useDeveloperMode,
  usePreventScreenCapture,
  useTour,
  walletTimeout,
}
export type { BannerMessage, DeepPartial, IButton }
