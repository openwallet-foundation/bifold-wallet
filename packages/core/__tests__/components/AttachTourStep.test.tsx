import { render } from '@testing-library/react-native'
import React from 'react'
import { View } from 'react-native'

import { AttachTourStep } from '../../src/components/tour/AttachTourStep'
import { TourProvider } from '../../src/contexts/tour/tour-provider'
import { tours } from '../../src/constants'
import { BaseTourID } from '../../src/types/tour'

describe('AttachTourStep Component', () => {
  test('Renders properly with defaults', () => {
    const tree = render(
      <TourProvider tours={tours} overlayColor={'gray'} overlayOpacity={0.7}>
        <AttachTourStep index={0} tourID={BaseTourID.HomeTour}>
          <View />
        </AttachTourStep>
      </TourProvider>
    )

    expect(tree).toMatchSnapshot()
  })
})
