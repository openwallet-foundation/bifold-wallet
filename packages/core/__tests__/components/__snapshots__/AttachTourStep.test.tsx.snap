// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttachTourStep Component Renders properly with defaults 1`] = `
<View
  collapsable={false}
  focusable={false}
  style={
    Object {
      "alignSelf": "center",
    }
  }
  testID="com.ariesbifold:id/AttachTourStep"
>
  <View />
</View>
`;
